packages/zarf-agent/manifests/webhook.yaml

```
@@ -91,3 +91,91 @@ webhooks:
       - "v1"
       - "v1beta1"
     sideEffects: None
+  - name: agent-argocd-application.zarf.dev
+    namespaceSelector:
+      matchExpressions:
+        # Ensure we don't mess with kube-system
+        - key: "kubernetes.io/metadata.name"
+          operator: NotIn
+          values:
+            - "kube-system"
+        # Allow ignoring whole namespaces
+        - key: zarf.dev/agent
+          operator: NotIn
+          values:
+            - "skip"
+            - "ignore"
+    objectSelector:
+      matchExpressions:
+        # Always ignore specific resources if requested by annotation/label
+        - key: zarf.dev/agent
+          operator: NotIn
+          values:
+            - "skip"
+            - "ignore"
+    clientConfig:
+      service:
+        name: agent-hook
+        namespace: zarf
+        path: "/mutate/argocd-application"
+      caBundle: "###ZARF_AGENT_CA###"
+    rules:
+      - operations:
+          - "CREATE"
+          - "UPDATE"
+        apiGroups:
+          - "argoproj.io"
+        apiVersions:
+          - "v1alpha1"
+        resources:
+          - "applications"
+    admissionReviewVersions:
+      - "v1"
+      - "v1beta1"
+    sideEffects: None
+  - name: agent-argocd-repository.zarf.dev
```

```suggestion
  - name: agent-argocd-repocreds.zarf.dev
```
It may be better to call these repo creds since that is what they are called in the upstream code (and they are distinct from `Repository`) - https://github.com/argoproj/argo-cd/blob/master/pkg/apis/application/v1alpha1/repository_types.go#L16

---

Technically they are not repocreds.  Repocreds are credential templates that enable for reuse across multiple Repositories.  We are indeed catching a Repository.

---

Ah I see the correct struct now: https://github.com/argoproj/argo-cd/blob/master/util/settings/settings.go#L286

---

Ah yeah here is the magic function: https://github.com/argoproj/argo-cd/blob/d7632df378577229c8228b6e2d4c8aeeb31e7f7e/util/db/repository_secrets.go#L302 (they don't just marshal it over like I expected)